### PR TITLE
Initial spec

### PR DESCRIPTION
**Pull request by jessealama**

### CODE REVIEW

File: spec.emu [hunk 1]
```diff
@@ -0,0 +1,205 @@
...
+      <emu-production name="SignedExponentialDigitString" type="lexical">
+	<emu-rhs>-? <emu-prodref name="ExponentialDigitStringLHS"/> ("e" | "E") <emu-prodref name="ExponentialDigitStringRHS"/></emu-rhs>
+      </emu-production>
+    </emu-grammar>
+    <emu-grammar name="infinity">
+      <emu-production>("I" | "i") ("N" | "n") | ("F" | "f") | ( ("I" | "i") | ("N" | "n") | ("I" | "i") | ("T" | "t") | ("Y" | "y") ) )?</emu-production>
+    </emu-grammar>
+    <emu-grammar name="nan">
+      <emu-production>("N" | "n") | ("A" | "a") | ("N" | "n")</emu-production>
+    </emu-grammar>
```
Comment (ptomato) [inline spec.emu:83]: I'm not sure that case-insensitive Infinity and NaN are the right thing to do here, but in any case a prose assertion would probably be clearer:
```
`Infinity` [> or any token that is an ASCII case-insensitive match]
```

**Review by ptomato — APPROVED** [review]
A few extra things I'd recommend doing even at this early stage, because you'll need to do them regardless of what shape the proposal takes in the end:

1. Make it easy to build a local version of the spec with ecmarkup, and set up linting.

The customary way to do this is with a `package.json` file. Here's one adapted from Temporal that you could use:
```json
{
  "name": "proposal-decimal",
  "version": "1.0.0",
  "description": "Representing base-10 decimal numbers",
  "scripts": {
    "build": "ecmarkup --lint-spec --strict --load-biblio @tc39/ecma262-biblio spec.emu out/index.html"
  },
  "repository": {
    "type": "git",
    "url": "git+https://github.com/tc39/proposal-decimal.git"
  },
  "keywords": [
    "TC39",
    "Decimal",
    "BigDecimal",
    "Decimal128"
  ],
  "author": "Jesse Alama <jesse@igalia.com>",
  "license": "SEE LICENSE IN LICENSE.md",
  "bugs": {
    "url": "https://github.com/tc39/proposal-decimal/issues"
  },
  "homepage": "https://tc39.es/proposal-decimal/",
  "devDependencies": {
    "@tc39/ecma262-biblio": "=2.1.2669",
    "ecmarkup": "^18.1.1"
  }
}
```
With this, anyone can check out the repo, run `npm install` once, and then `npm run build` to build a local copy of the spec in `out/index.html`. You will probably find that it's easier to get contributions from others when it's set up this way.

2. I'd recommend using [StringToNumber](https://tc39.es/ecma262/#sec-stringtonumber) as a basis for the string-to-decimal algorithm.

3. Last, I'd recommend setting up CI to not just build the spec, but also publish it, so that you don't have to commit the HTML file to the repo. Here's an [example workflow file](https://github.com/tc39/proposal-temporal/blob/1a18bc08d0e75b0b25b7ae57f37c5521ee3a5ed1/.github/workflows/deploy.yml) from Temporal

File: spec.emu [hunk 2]
```diff
@@ -0,0 +1,205 @@
...
+      <emu-production name="DigitString" type="lexical">
+	<emu-rhs><emu-prodref name="DecimalDigit"/>+</emu-rhs>
+      </emu-production>
+      <emu-production name="UnsignedDecimalDigitString" type="lexical" oneof>
+	<emu-rhs><emu-prodref name="DigitString"/></emu-rhs>
+	<emu-rhs><emu-prodref name="DigitString"/> "." <emu-prodref name="DigitString"/></emu-rhs>
+      </emu-production>
+      <emu-production name="SignedDecimalDigitString" type="lexical" oneof>
+	<emu-rhs>-? <emu-prodref name="UnsignedDigitString"/></emu-rhs>
+      </emu-production>
```
[the remark this answers is not in the document]
Comment (jessealama) [reply spec.emu]: Ah, of course, thanks!

I think supporting digit strings with a separator is a good idea, to make these strings compatible with the familiar JS numeric literals. And even if that is an extension of the official IEEE 754 Decimal128 syntax, that's fine. It's harmless in terms of the semantics, of course. It's a small price to pay.

[the remark this answers is not in the document]
Comment (jessealama) [reply spec.emu]: Yes, I think we should support things like `+123.456`. JS supports that notation.

One difficulty there, though, is that round-trippability from strings to Decimal objects and back to strings could fail. At the moment, we envision a `toString` method that takes no parameters, and internally we track only whether a number is negative, so there would be no basis for emitting a leading `+` in the `toString`.

File: spec.emu (hunk 1 above)
Comment (jessealama) [reply spec.emu:74]: Agree. Unfortunately IEEE 754 allows any case, so this is done to be compatible with that.

File: spec.emu [hunk 3]
```diff
@@ -0,0 +1,205 @@
...
+    <p>Decimal values defined in this section are <emu-xref href="#ieee-754-2019">IEEE-754 Decimal128 values</emu-xref>. The full spectrum of values defined by Decimal128 are available here, including NaN and positive and negative infinity. (NB: The values NaN and both infinities, available here, are distinct from the built-in NaN, *Infinity* and *-Infinity* values that arise from the 64-bit binarry IEEE arithmetic used outside of the Decimal universe.) Decimal128 is a floating-point format whose range of values allows up to 34 significant digits and an exponent ranging from -6143 to 6144.</p>
+  </emu-intro>
+  <emu-clause id="sec-the-decimal-constructor">
+    <h1>The Decimal Constructor</h1>
+    <p>The constructor for the Decimal object takes a single argument. If the argument is not a String, throw a *TypeError*. The String argument must match one of the following four valid syntaxes:</p>
+    <emu-grammar name="decimal-syntax">
```
[the remark this answers is not in the document]
Comment (jessealama) [reply spec.emu:40]: That sounds good, thanks! I'll work on that in a future PR.

File: spec.emu [hunk 4]
```diff
@@ -0,0 +1,205 @@
...
+    </emu-grammar>
+    <emu-grammar>
+      <emu-production name="UnsignedDecimalDigitString" type="lexical" oneof>
+	<emu-rhs><emu-prodref name="DigitString"/></emu-rhs>
+	<emu-rhs><emu-prodref name="DigitString"/> "." <emu-prodref name="DigitString"/></emu-rhs>
+      </emu-production>
+      <emu-production name="SignedDecimalDigitString" type="lexical" oneof>
+	<emu-rhs>-? <emu-prodref name="UnsignedDigitString"/></emu-rhs>
+      </emu-production>
+    </emu-grammar>
```
[the remark this answers is not in the document]
Comment (jessealama) [reply spec.emu]: Oops, you're right. Thanks!

**Pull request merged**